src/linter/linter.ts

```diff
@@ -11,8 +11,47 @@ import type {AbstractReader, Resource} from "@ui5/fs";
 import ConfigManager, {UI5LintConfigType} from "../utils/ConfigManager.js";
 import {Minimatch} from "minimatch";
 
+async function lint(
+	resourceReader: AbstractReader, options: LinterOptions, config: UI5LintConfigType
+): Promise<LintResult[]> {
+	const lintEnd = taskStart("Linting");
+	let {ignorePattern, filePatterns} = options;
+	const {rootDir} = options;
+
+	// Resolve files to include
+	filePatterns = [
+		...(config.files ?? []),
+		...(filePatterns ?? []), // CLI patterns go after config patterns
```

Comment:
I'm not sure whether this is the behavior a user would expect. If the config file specifies `webapp/**/*`, and the user provides `webapp/controller/BaseController.js` via the CLI, the current implementation will still lint everything in `webapp`. I think the user would rather expect to only lint `webapp/controller/BaseController.js`.

What do you think? Maybe we should ignore the config file if filePatterns are provided?

Reply:
Yep, makes totally sense.
Maybe, I have been confused by the ignores property where we merge them, but the CLI has precedence.
I'll refactor it!

Thanks!